components/clp-credential-manager/src/config.rs

```diff
@@ -0,0 +1,113 @@
+use std::{fs, net::SocketAddr, path::Path};
+
+use secrecy::SecretString;
+use serde::Deserialize;
+
+use crate::error::{ServiceError, ServiceResult};
+
+const DEFAULT_MAX_CONNECTIONS: u32 = 5;
+const DEFAULT_SERVER_BIND_ADDRESS: &str = "0.0.0.0";
+const DEFAULT_SERVER_PORT: u16 = 8080;
+
+/// Root configuration deserialized from `credential-manager-config.yml`.
+#[derive(Debug, Clone, Deserialize)]
+pub struct AppConfig {
+    #[serde(default)]
+    pub server: ServerConfig,
+    pub database: DatabaseConfig,
+}
+
+impl AppConfig {
+    /// Loads configuration from disk and validates YAML syntax.
+    ///
+    /// # Parameters:
+    ///
+    /// * `path`: Filesystem location of the YAML configuration file.
+    ///
+    /// # Returns:
+    ///
+    /// A parsed [`AppConfig`] instance when the file can be read and deserialized successfully.
+    ///
+    /// # Errors:
+    ///
+    /// * Returns [`ServiceError::Io`] if the file cannot be read.
+    /// * Returns [`ServiceError::Yaml`] if parsing fails.
+    pub fn from_file(path: &Path) -> ServiceResult<Self> {
+        let contents = fs::read_to_string(path)?;
+        let config: Self = serde_yaml::from_str(&contents)?;
+        Ok(config)
+    }
+}
+
+/// Network settings for the Axum server.
+#[derive(Debug, Clone, Deserialize)]
+pub struct ServerConfig {
+    #[serde(default = "default_bind_address")]
+    pub bind_address: String,
+    #[serde(default = "default_bind_port")]
+    pub port: u16,
+}
```

Comment:
```suggestion
#[serde(default)]
pub struct ServerConfig {
    pub bind_address: String,
    pub port: u16,
}
```
We actually don't need standalone functions such as `default_bind_address` and `default_bind_port` if we have `impl Default for ServerConfig`